libvast/vast/detail/assert.hpp

```diff
@@ -14,31 +14,18 @@
 #pragma once
 
 #include "vast/config.hpp"
+#include "vast/detail/backtrace.hpp"
 
 #if VAST_ENABLE_ASSERTIONS
-#  if VAST_WINDOWS
-#    include <cstdio>
-#    include <cstdlib>
-#    define VAST_ASSERT(expr)                                                  \
-      if (static_cast<bool>(expr) == false) {                                  \
-        ::printf("%s:%u: assertion failed '%s'\n", __FILE__, __LINE__, #expr); \
-        ::abort();                                                             \
-      }                                                                        \
-      static_cast<void>(0)
-#  else // VAST_LINUX || VAST_MACOS || VAST_BSD
-#    include <cstdio>
-#    include <cstdlib>
-#    include <execinfo.h>
-#    define VAST_ASSERT(expr)                                                  \
-      if (static_cast<bool>(expr) == false) {                                  \
-        ::printf("%s:%u: assertion failed '%s'\n", __FILE__, __LINE__, #expr); \
-        void* vast_array[10];                                                  \
-        auto vast_bt_size = ::backtrace(vast_array, 10);                       \
-        ::backtrace_symbols_fd(vast_array, vast_bt_size, 2);                   \
-        ::abort();                                                             \
-      }                                                                        \
-      static_cast<void>(0)
-#  endif
+#  include <cstdio>
+#  include <cstdlib>
+#  define VAST_ASSERT(expr)                                                    \
+    if (static_cast<bool>(expr) == false) {                                    \
+      ::printf("%s:%u: assertion failed '%s'\n", __FILE__, __LINE__, #expr);   \
+      vast::detail::backtrace();                                               \
+      ::abort();                                                               \
+    }                                                                          \
+    static_cast<void>(0)
```

Comment:
Instead of doing this, can you wrap the whole macro content in `do { .... } while(false)` so it's a single expression?